### PR TITLE
Add FSM based CCE implementation, fixup all ME tests

### PR DESCRIPTION
**Pull request by muwyse**

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -70,3 +70,8 @@
   branch = blackparrot_mods
   ignore = dirty
 
+[submodule "external/cmurphi"]
+	path = external/cmurphi
+	url = git@github.com:black-parrot/cmurphi.git
```
Comment (dpetrisko) [inline .gitmodules]: Change this to https, otherwise it sometimes fails if there's no github username

File: bp_common/syn/Makefile.vcs [hunk 2]
```diff
@@ -61,17 +61,19 @@ lint.v: build.v
 BUILD_LOG    ?= $(LOG_DIR)/$(TB).$(CFG).build.log
 BUILD_REPORT ?= $(REPORT_DIR)/$(TB).$(CFG).build.rpt
 BUILD_ERROR  ?= $(REPORT_DIR)/$(TB).$(CFG).build.err
-build.v: dirs.v 
+pre-build.v: dirs.v
```
Comment (dpetrisko) [inline bp_common/syn/Makefile.vcs]: good change!

File: bp_me/test/tb/bp_cce/Makefile.frag [hunk 3]
```diff
@@ -0,0 +1,15 @@
+AXE_TRACE_P ?= 0
+CCE_TRACE_P  ?= 0
+SKIP_INIT_P ?= 0
+LCE_PERF_TRACE_P ?= 0
+
+DUT_PARAMS = -pvalue+axe_trace_p=$(AXE_TRACE_P) \
+             -pvalue+cce_trace_p=$(CCE_TRACE_P) \
+						 -pvalue+instr_count=$(NUM_INSTR_P) \
+						 -pvalue+skip_init_p=$(SKIP_INIT_P) \
```
Comment (dpetrisko) [inline bp_me/test/tb/bp_cce/Makefile.frag]: watch the tabs here

File: bp_top/test/tb/bp_top_trace_demo/Makefile.frag [hunk 4]
```diff
@@ -1,8 +1,11 @@
 CCE_TRACE_P  ?= 0
 CALC_TRACE_P ?= 0
+SKIP_INIT_P  ?= 0
 
 DUT_PARAMS = -pvalue+calc_trace_p=$(CALC_TRACE_P) \
-             -pvalue+cce_trace_p=$(CCE_TRACE_P)
+             -pvalue+cce_trace_p=$(CCE_TRACE_P)   \
+						 -pvalue+skip_init_p=$(SKIP_INIT_P)
```
Comment (dpetrisko) [inline bp_top/test/tb/bp_top_trace_demo/Makefile.frag]: tab

**Comment by taylor-bsg** [comment]
Cool!!!!

**Comment by muwyse** [comment]
Reviews have been addressed

**Comment by muwyse** [comment]
regress.v, regress_beebs.v and hello_world all run successfully

atomic_queue_demo runs for 2, 4, 8, and 16 cores

**Pull request merged**